qiskit_dynamics/signals/signals.py

```diff
@@ -445,24 +421,24 @@ def add_samples(self, start_sample: int, samples: List):
         Raises:
             QiskitError: If start_sample is less than the current length of samples.
         """
-        samples = Array(samples)
+        samples = unp.asarray(samples)
 
         if len(samples) < 1:
             return
 
         if start_sample < len(self.samples):
             raise QiskitError("Samples can only be added afer the last sample.")
 
-        zero_pad = np.expand_dims(np.zeros_like(Array(samples[0])), 0)
+        zero_pad = unp.expand_dims(unp.zeros_like(unp.asarray(samples[0])), 0)
```

Comment:
Same as a previous comment, try changing this to:
```suggestion
        zero_pad = unp.expand_dims(unp.zeros_like(samples[0]), 0)
```